em_workflows/czi/flow.py

```diff
@@ -1,28 +1,85 @@
+from pytools.HedwigZarrImages import HedwigZarrImages
 from em_workflows.file_path import FilePath
 from em_workflows.utils import utils
-from em_workflows.config import Config
 from prefect.run_configs import LocalRun
 from .constants import VALID_CZI_INPUTS
+from .config import CZIConfig
+
+
+def rechunk_zarr(zarr_fp: Path) -> None:
+    images = HedwigZarrImages(zarr_fp, read_only=False)
+    for _, image in images.series():
+        image.rechunk(512)  # config.CHUNK_SIZE
+
+
+def gen_thumb(image: HedwigZarrImage, file_path: FilePath) -> dict:
+    sitk_image = image.extract_2d(target_size_x=300, target_size_y=300, auto_uint8=True)
+    if sitk_image:
+        output_jpeg = f"{file_path.working_dir}/{file_path.base}_sm.jpeg"
+        sitk.WriteImage(
+            sitk_image,
+            output_jpeg,
+            useCompression=True,
+            compressionLevel=90,
+        )
+        asset_fp = file_path.copy_to_assets_dir(fp_to_cp=Path(output_jpeg))
+        thumb_asset = file_path.gen_asset(asset_type="thumbnail", asset_fp=asset_fp)
+        return thumb_asset
+
+
+def gen_imageSet(file_path: FilePath) -> List:
+
+    zarr_fp = f"{file_path.assets_dir}/{file_path.base}.zarr"
+    imageSet = list()
+    zarrImages = HedwigZarrImages(Path(zarr_fp))
+    for image_name, image in zarrImages.series():
+        # single image element
+        image_elt = dict()
+        image_elt["imageMetadata"] = None
+        assets = list()
+        if image_name == "macro image":
+            # we don't care about the macro image
+            continue
+        assets.append(gen_thumb(image=image, file_path=file_path))
+        image_elt["imageName"] = image_name
+
+        if image_name != "label image":
+            ng_asset = file_path.gen_asset(
+                asset_type="neuroglancerZarr", asset_fp=Path(zarr_fp)
+            )
+            ng_asset["metadata"] = dict(
+                shader=image.shader_type,
+                dimensions=image.dims,
+                shaderParameters=image.neuroglancer_shader_parameters(),
+            )
+            assets.append(ng_asset)
+        image_elt["assets"] = assets
+        imageSet.append(image_elt)
+    return imageSet
 
 
 @task
 def bioformats_gen_zarr(file_path: FilePath):
     """
+    TODO sort ot the num workers
     TODO, refactor this into ng.gen_zarr
     bioformats2raw --max_workers=$nproc  --downsample-type AREA
     --compression=blosc --compression-properties cname=zstd
     --compression-properties clevel=5 --compression-properties shuffle=1
     input.tiff output.zarr
     """
 
-    input_czi = f"{file_path.working_dir}/{file_path.base}.czi"
+    input_czi = f"{file_path.proj_dir}/{file_path.base}.czi"
     output_zarr = f"{file_path.working_dir}/{file_path.base}.zarr"
     log_fp = f"{file_path.working_dir}/{file_path.base}_as_zarr.log"
     cmd = [
-        Config.bioformats2raw,
-        "--max_workers=8",
+        CZIConfig.bioformats2raw,
+        "--max_workers=19",
```

Comment:
Can you elaborate in a comment about why you chose 19 as the number of workers?

Reply:
let's make it 20 for now, and config this number.
I think even at 20 we're seeing drop off in perf gains, for the tests this seems ok? But we can tune this over time.
I'm not sure that simply using all procs is a good idea here. We can put a note that says this could be tuned.